### PR TITLE
Feature/implement accessibility checks

### PR DESCRIPTION
**Pull request by ElRaptorus**
Depends on:
https://github.com/process-engine/iam/pull/4
https://github.com/process-engine/process_engine_contracts/pull/40
https://github.com/process-engine/process_engine/pull/110
https://github.com/process-engine/consumer_api_core/pull/40

## What did you change?

Prepare the integrationtests for usage of the identity server and the new iam concept:
- Rename all process model lanes to `Default_Test_Lane`, which equals a test claim defined in the identity service
- Refactor the context generation
  - Instead of storing fixtures in the PostgresDB and creating an Execution- and Consumer- Context based on those, the FixtureProvider will perform a request against the identity server before starting the tests
  - The request itself is handled by a login-provider, which lies in the same directory as the fixture provider
  - **NOTE**: As a result of this, the tests will no longer work, unless you have an instance of the [identity server](https://github.com/process-engine/identity_server) running on your machine!
  - Also note that, as a side effect of moving all auth-related checks to an external authority, the tests will now take considerably longer to run

TODO:
- The tests for the `TerminateEndEvent` do not yet use the new handlers and are, for now, not executable
See #89 
This is best fixed, when the new persistence layer has been implemented.

## How can others test the changes?

- Start up an identity server from the linked repository
- Run the tests

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

**Comment by ElRaptorus** [comment]
https://github.com/process-engine-pro/organization/issues/150

File: _integration_tests/src/iam_facade_mock.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import {IIdentity} from '@essential-projects/iam_contracts';
+import {IIamFacade} from '@process-engine/process_engine_contracts';
+
+export class IamFacadeMock implements IIamFacade {
+
+  public async checkIfUserCanAccessLane(identity: IIdentity, laneId: string): Promise<boolean> {
+    return Promise.resolve(true);
```
Comment (cmg-dev) [inline _integration_tests/src/iam_facade_mock.ts]: ich verstehe die IAM Facade nicht... warum sehe ich nicht den IamService als Mock?

File: _integration_tests/src/index.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+export * from './iam_facade_mock';
```
Comment (cmg-dev) [inline _integration_tests/src/index.ts]: iam_service_mock hätte ich erwartet.

File: _integration_tests/src/test_fixture_provider.ts [hunk 3]
```diff
@@ -1,17 +1,14 @@
...
-// tslint:disable-next-line:import-blacklist
-import * as _ from 'lodash';
 import * as path from 'path';
 
 import {InvocationContainer} from 'addict-ioc';
 import {Logger} from 'loggerhythm';
 
-import {ExecutionContext, IIamService, TokenType} from '@essential-projects/core_contracts';
-import {IProcessDefEntityTypeService, IProcessEngineService, IProcessRepository} from '@process-engine/process_engine_contracts';
+import {ExecutionContext, IProcessDefEntityTypeService, IProcessEngineService, IProcessRepository} from '@process-engine/process_engine_contracts';
```
Comment (cmg-dev) [inline _integration_tests/src/test_fixture_provider.ts]: formatieren

File: _integration_tests/src/test_fixture_provider.ts [hunk 4]
```diff
@@ -210,37 +200,24 @@ export class TestFixtureProvider {
...
-  private async createConsumerContext(user: string, password: string): Promise<ConsumerContext> {
-    const authToken: any = await this.bootstrapper.getTokenFromAuth(user, password);
+  private _createMockContexts(): void {
+
+    // Note: Since the iam service is mocked, it doesn't matter what kind of token is used here.
+    // It only matters that one is present.
+    const identity: IIdentity = {
+      token: 'randomtoken',
+    };
+    this._executionContext = new ExecutionContext(identity);
```
Comment (cmg-dev) [inline _integration_tests/src/test_fixture_provider.ts]: aber auch das hätte ich durch den IdentityService gegeben.

Von dem bekomme ich eine Identity.

Die schiebe ich in den IamService (Mock) rein, der dann einfach OK sagt.

Das kann ich dann super einfach, gegen einen Test tauschen, der gegen den IdentityServer geht, da ich nur den Mock durch die richtige Implementierung tauschen muss.

File: _integration_tests/src/iam_facade_mock.ts (hunk 1 above)
Comment (ElRaptorus) [reply _integration_tests/src/iam_facade_mock.ts]: Die Iam Fassade benutzt im nicht-gemockten Zustand den Iam Service.

Die hatte ich eingezogen, um Stellen wie diese hier etwas eleganter gestalten zu können.
https://github.com/process-engine/process_engine/blob/3c7ab8cf562e096b3fe0e9ba91d10752e233771b/src/new_model/runtime/storage/service/process_model_persistence_service.ts#L115

File: _integration_tests/src/index.ts (hunk 2 above)
Comment (ElRaptorus) [reply _integration_tests/src/index.ts]: siehe oben.

File: _integration_tests/src/test_fixture_provider.ts (hunk 4 above)
Comment (ElRaptorus) [reply _integration_tests/src/test_fixture_provider.ts]: Das was hier erzeugt wird, ist doch eine Identity? 
Verstehe grade das Problem nicht ganz.

File: _integration_tests/src/iam_facade_mock.ts (hunk 1 above)
Comment (5ebastianMeier) [reply _integration_tests/src/iam_facade_mock.ts]: Damit die Kapselung klarer wird sollten wir das vielleicht `ProcessModelIamFacade` oder `ProcessModelIamService` nennen.

Dann müsste für einen Test allerdings wirklich der darin verwendete `IamService` gemockt werden, da wir ansonsten die Logik der Fassade vom Test ausnehmen würden.

Comment (ElRaptorus) [reply _integration_tests/src/iam_facade_mock.ts]: @5ebastianMeier @cmg-dev 
Hab die Fassade rausgeschmissen. War doof. Merk ich selbst. 😢

**Pull request merged**